src/tests/ftest/get_remote_files.sh

```diff
@@ -0,0 +1,369 @@
+    for file in $(ls -d ${1})
+    do
+        file_name=${file##*/}
+        archive_name="${file_name%%.*}.$(hostname -s).${file_name#*.}"
+        if scp -r "${file}" "${2}"/"${archive_name}"; then
+            copied+=("${file}")
+            if ! rm -fr "${file}"; then
+                echo "  Error removing ${file}"
+                echo "    $(ls -al ${file})"
+                rc=1
+            fi
+        else
+            echo "  Failed to archive ${file} to ${2}"
+            rc=1
+        fi
+    done
+    echo "  The following files were archived:"
+    for file in "${copied[@]:-no files}"
+    do
+        echo "    ${file}"
+    done
+    return ${rc}
+}
+
+#######################################
+# Run cart_logtest.py against the logs.
+# Arguments:
+#   $1: local logs
+# Returns:
+#   0: All cart_logtest were successful
+#   1: Failure detected with cart_logtest
+#######################################
+run_cartlogtest() {
+    rc=0
+    if [ -f "${CART_LOGTEST_PATH}" ]; then
+        # shellcheck disable=SC2045,SC2086
+        for file in $(ls -d ${1})
+        do
+            if [ -f ${file} ]; then
+                logtest_log="${file%.*}.cart_logtest.${file##*.}"
+                ${CART_LOGTEST_PATH} ${file} > ${logtest_log} 2>&1
+                ret=$?
+                if [ ${ret} -ne 0 ]; then
+                    echo "  Error: details in ${file}_cart_testlog"
+                    rc=1
+                fi
+            fi
+        done
+    else
+        echo "  Error: ${CART_LOGTEST_PATH} does not exist!"
+        rc=1
+    fi
+    return ${rc}
+}
+
+####### MAIN #######
+if [[ $# -eq 0 ]] ; then
+    usage
+    exit 1
+fi
+
+# Setup defaults
+ARCHIVE_DEST=""
+FILES_TO_PROCESS=""
+COMPRESS="false"
+VERBOSE="false"
+CART_LOGTEST="false"
+EXCLUDE_ZIP="false"
+THRESHOLD=""
+rc=0
+
+# Step through arguments
+while getopts "vhxzca:f:t:" opt; do
+    case ${opt} in
+        a )
+            ARCHIVE_DEST=$OPTARG
+            ;;
+        f )
+            FILES_TO_PROCESS=$OPTARG
+            ;;
+        t )
+            THRESHOLD=$OPTARG
+            ;;
+        c )
+            CART_LOGTEST="true"
+            ;;
+        z )
+            COMPRESS="true"
+            ;;
+        x )
+            EXCLUDE_ZIP="true"
+            ;;
+        v )
+            VERBOSE="true"
+            ;;
+        h )
+            usage
+            exit
+            ;;
+        ? )
+            echo "Unexpected argument: ${opt}!"
+            usage
+            exit
+            ;;
+    esac
+done
+shift "$((OPTIND -1))"
+
+script_name=$(basename "${0}")
+log_file="/tmp/${script_name%%.*}_$(/bin/date +%Y%m%d_%H%M%S).log"
+if [ "${VERBOSE}" != "true" ]; then
+    # Redirect 'set -e' output to a log file if not verbose
+    exec 2> "${log_file}"
+fi
+set -ex
+echo "Logging ${script_name} output to ${log_file}"
+
+# Verify files have been specified and they exist on this host
+check_files_input "${FILES_TO_PROCESS}"
```

Comment:
This won't work with `set -e` enabled as the script will exit if `check_files_input` exits `!0`.  It should at least exit with an error message:

```suggestion
if ! check_files_input "${FILES_TO_PROCESS}"; then
    echo ...
    exit 1
fi
```

Reply:
Will fix

Reply:
We are hitting the scenario described by Brian currently:
```
[2020-10-17T08:06:47.843Z] Running on wolf-106vm[1-6]: /usr/lib/daos/TESTING/ftest/get_remote_files.sh -z -a "wolf-106vm1:/var/tmp/ftest/avocado/job-results/latest/cart_logs" -f "/var/tmp/daos_testing/*/*log*" -c -t "1G"
[2020-10-17T08:06:47.843Z]   Errors detected running "archive_cart_logs command":
[2020-10-17T08:06:47.843Z]     wolf-106vm[1-6]: rc=2, output: Logging get_remote_files.sh output to /tmp/get_remote_files_20201017_080641.log
```
Would a `case` statement work in this case with `set -e`?
```
case check_files_input "${FILES_TO_PROCESS}" in
1)
    echo "Please specify -f option."
    exit 1
    ;;
2)
    echo "No files matched ${FILES_TO_PROCESS}. Nothing to do."
    exit 0
    ;;
esac
```